### PR TITLE
add validation error summary

### LINKED ISSUE #144: tools/validate: add a makeshift summary report
- output a new file `summary.txt` with counts for all individual errors, per table
- print the total number of errors per table at the end of program execution, as well as a reminder of where to find the output files

### PR DESCRIPTION
**Pull request by zargot**
fixes #144

### CODE REVIEW

File: src/odm_validation/validation.py [hunk 1]
```diff
@@ -223,7 +225,12 @@ def _strip_coerce_rules(cerb_schema):
     return strip_dict_key(deepcopy(cerb_schema), 'coerce')
 
 
-def _map_cerb_errors(cerb_errors, schema, rule_whitelist, offset: int):
+def _record_error(summary: reports.ValidationSummary, table_id, rule_id):
```
Comment (yulric) [inline src/odm_validation/validation.py:228]: Should move this function to the `ValidationSummary` class? And rename it to `_update_report`?

File: tools/validate_xlsx.py [hunk 2]
```diff
@@ -41,12 +43,40 @@ def get_sheet_table_id(schema, sheet_name) -> Optional[str]:
...
+def gen_summary(summary: reports.ValidationSummary) -> Optional[str]:
+    result = '# error summary\n'
+    total = 0
+    for table_id, table_summary in summary.table_summaries.items():
+        result += f'\n## {table_id}\n'
+        table_count_max = reduce(lambda a, b: max(a, b),
+                                 table_summary.error_counts.values())
+        width = len(str(table_count_max))
+        for rule_id, count in table_summary.error_counts.items():
+            result += f'- {count:{width}d} {rule_id}\n'
```
Comment (yulric) [inline tools/validate_xlsx.py:75]: Add a comment here about why the width is needed.

**Review by yulric — CHANGES_REQUESTED** [review]
1. Think we should move all the logic for creating the summary report into a seperate function, in prepearation for the new create_summary_report function that we discussed before.
2. Looking at the new code that requires passing in the ruleId is making me think we should  normalize the Error object so that it should have an error type field (Warning or Error) and a ruleId field. That way you don't have to pass it in everywhere and just get it from the object.

File: tools/validate_xlsx.py (hunk 2 above)
Comment (zargot) [reply tools/validate_xlsx.py:75]: done

File: src/odm_validation/validation.py (hunk 1 above)
Comment (zargot) [reply src/odm_validation/validation.py:228]: done, i called it count_error

**Review by yulric — APPROVED** [review]

**Pull request merged**